include/llbuild/BuildSystem/CommandResult.h

```diff
@@ -34,20 +34,21 @@ struct CommandExtendedResult {
   uint64_t utime;       /// User time (in us)
   uint64_t stime;       /// Sys time (in us)
   uint64_t maxrss;      /// Max RSS (in bytes)
+  int32_t pid;          /// Process identifier (can be -1 for failure reasons)
 
 
-  CommandExtendedResult(CommandResult result, int exitStatus, uint64_t utime = 0,
+  CommandExtendedResult(CommandResult result, int exitStatus, int32_t pid, uint64_t utime = 0,
                 uint64_t stime = 0, uint64_t maxrss = 0)
-    : result(result), exitStatus(exitStatus)
+    : result(result), exitStatus(exitStatus), pid(pid)
     , utime(utime), stime(stime), maxrss(maxrss)
   {}
 
-  static CommandExtendedResult makeFailed(int exitStatus = -1) {
-    return CommandExtendedResult(CommandResult::Failed, exitStatus);
+  static CommandExtendedResult makeFailed(int32_t pid, int exitStatus = -1) {
```

Comment:
Might read a bit cleaner if we put the value for 'no pid' here as the default value, that way callers don't need to be aware of what that value needs to be.